### ISSUE TITLE
Clarifying local descriptors in documentation

### ISSUE THREAD

**Issue by scmartin** [glotzerlab/freud#1026]
### Description

The `environment.LocalDescriptors` documentation doesn't really give a description of what the local descriptors are, how they are calculated, or even a reference that explains what they are. As far as I can tell, one would have to actually look through the source code to find out what this class actually calculates. In a previous version of the documentation there was [this example](https://freud.readthedocs.io/en/v2.0.0/examples/module_intros/environment.LocalDescriptors.html) but it is no longer in the 'stable' branch. The documentation for this class should be more explicit about what it calculates and what its properties are.

### Motivation and Context

As an end user, it is hard to know whether and how to use classes like this without more information about what they are calculating.

**Comment by bdice** [comment]
I agree this documentation (https://freud.readthedocs.io/en/latest/modules/environment.html#freud.environment.LocalDescriptors) can be improved. The publication that developed and used this feature is "Machine learning for crystal identification and discovery" by Matthew Spellings and Sharon C. Glotzer. https://doi.org/10.1002/aic.16157 cc: @klarh

**Comment by bdice** [comment]
@scmartin The example you mentioned has changed URLs, but is still a part of the docs.
- Latest: https://freud.readthedocs.io/en/latest/gettingstarted/examples/module_intros/environment.LocalDescriptors.html
- v2.11.0 (latest at time of writing): https://freud.readthedocs.io/en/v2.11.0/gettingstarted/examples/module_intros/environment.LocalDescriptors.html
